test/linter/test-versions.js

```diff
@@ -72,6 +72,48 @@ function removedAfterAdded(statement) {
   );
 }
 
+/**
+ * @param {VersionValue} xver
+ * @param {VersionValue} yver
+ */
+function compareVersionNumber(xver, yver) {
+  /**
+   * @param {VersionValue} version
+   */
+  function toArray(version) {
+    if (typeof version === 'string') {
+      return (version.startsWith('≤') ? version.slice(1) : version).split('.');
+    }
+    return [0];
+  }
+  const x = toArray(xver);
+  const y = toArray(yver);
+  const length = Math.max(x.length, y.length);
+  for (let i = 0; i < length; i++) {
+    return Math.sign((x[i] || 0) - (y[i] || 0));
+  }
+  return 0;
+}
+
+/**
+ * @param {SimpleSupportStatement[]} supportDataList
+ * @param {string} relPath
+ * @param {Logger} logger
+ */
+function checkVersionOrder(supportDataList, relPath, browser, logger) {
```

Comment:
I think there are cases where this lint will require the wrong thing. For example here:

https://github.com/mdn/browser-compat-data/blob/445710d210ebfb33755b3e53f15918cb53890b8d/api/Element.json#L4501-L4517

The `webkitMatchesSelector` entry is the newest, but we don't want it to be the first one listed. Or maybe we should want that, but then we need to change the rendering rules of MDN, since that will use the first entry.